### ISSUE TITLE
consider renaming multipleOf to step (like the html-attribute)

### ISSUE THREAD

**Issue by nuxodin** [json-schema-org/json-schema-spec#905]
multipleOf is the equivalent of the step-attribute from html inputs.
Perhaps one could consider renaming this property?

Pros: 
- `step` is smaller
- the HTML standard is more established
- a step towards uniformity 

Cons:
- its a breaking change
- Is it less descriptive?

**Comment by notEthan** [comment]
the pros seem dubious at best. a shorter string isn't too compelling when it comes at the cost of describing what the keyword actually does.

uniformity with an html &lt;input&gt; tag seems ... irrelevant? moreover, 'step' doesn't even mean 'multiple of' in html inputs; it is only the increment. if you have a step of 3, when you start with a value of 1, incrementing goes to 4. neither is a multiple of 3.

**Comment by handrews** [comment]
@nuxodin at this point `multipleOf` has been a stable part of the spec for many years.  We don't just rename keywords (as you note, this is a breaking change) without compelling reasons.

We have (with `dependencies`) and probably will again (with `items`) split keywords that were doing too  many things at once.  And we renamed `definitions` to establish a clear core vocabulary (but because of the nature of that keyword you can keep using the old name without causing problems, so it's less breaking that it seems).

There are several keywords that have sub-optimal naming from one perspective or another, but we just have to deal with that.  That's the nature of broadly implemented projects.

**Comment by gregsdennis** [comment]
To add another nail to this coffin, it seems to me that `step` infers an iterative process, whereas `multipleOf` is just talking about numbers.

**Comment by awwright** [comment]
We already renamed this keyword from "divisibleBy", remember that? I think we changed it because it wasn't exactly accurate. However I think "multipleOf" is much more descriptive than "step".

This isn't a bad idea, by any means. And if there's good reasons to adopt the HTML behavior too ("step" checks that the value of the number instance, minus the "minimum" or "exclusiveMinimum", is a multiple of the "step") it could be a good excuse for a new keyword. I know there've been a few people asking "how do I ensure a number is odd" and the only way to do that is with "not".

Also, I think anyOf/allOf/oneOf should be named some/every/one (to match ECMAScript Array methods), but it's not really feasible to do at this point.

So, I think we've already been down this road for this keyword, but if the problem is that "divisibleBy" is _functionally_ insufficient, that could be something to look at.
